pkg/pdapi/pd_control.go

```diff
@@ -102,11 +104,41 @@ func (pdc *defaultPDControl) GetPDClient(namespace Namespace, tcName string, tls
 	return pdc.pdClients[key]
 }
 
+func (pdc *defaultPDControl) GetClusterRefPDClient(namespace Namespace, tcName string, clusterDomain string, tlsEnabled bool) PDClient {
+	pdc.mutex.Lock()
+	defer pdc.mutex.Unlock()
+
+	var tlsConfig *tls.Config
+	var err error
+	var scheme = "http"
+
+	if tlsEnabled {
+		scheme = "https"
+		tlsConfig, err = GetTLSConfig(pdc.kubeCli, namespace, tcName, util.ClusterClientTLSSecretName(tcName))
+		if err != nil {
+			klog.Errorf("Unable to get tls config for tidb cluster %q, pd client may not work: %v", tcName, err)
+			return &pdClient{url: ClusterRefPDClientUrl(namespace, tcName, scheme, clusterDomain), httpClient: &http.Client{Timeout: DefaultTimeout}}
+		}
+
+		return NewPDClient(ClusterRefPDClientUrl(namespace, tcName, scheme, clusterDomain), DefaultTimeout, tlsConfig)
+	}
+
+	key := ClusterRefpdClientKey(scheme, namespace, tcName, clusterDomain)
+	if _, ok := pdc.pdClients[key]; !ok {
+		pdc.pdClients[key] = NewPDClient(ClusterRefPDClientUrl(namespace, tcName, scheme, clusterDomain), DefaultTimeout, nil)
+	}
+	return pdc.pdClients[key]
+}
+
 // pdClientKey returns the pd client key
 func pdClientKey(scheme string, namespace Namespace, clusterName string) string {
 	return fmt.Sprintf("%s.%s.%s", scheme, clusterName, string(namespace))
 }
 
+func ClusterRefpdClientKey(scheme string, namespace Namespace, clusterName string, clusterDomain string) string {
+	return fmt.Sprintf("%s.%s.%s.%s", scheme, clusterName, string(namespace), clusterDomain)
```

Comment:
The `namespace` should not be empty here so we may have to set the `namespace` before calling this function.

Reply:
Is it necessary to check `clusterDomain`?

Reply:
fixed